.circleci/config.yml

```diff
@@ -0,0 +1,240 @@
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+              "$CIRCLE_SHA1" \
+              "$CIRCLE_TAG" \
+              "$CIRCLE_PROJECT_USERNAME" \
+              "$CIRCLE_PROJECT_REPONAME" \
+              "$CIRCLE_BUILD_URL" > src/backend/version.json
+      # Each image is tagged with the current git commit sha1 to avoid
+      # collisions in parallel builds.
+      - run:
+          name: Build production image
+          command: BUILD_TARGET=production make build
+      - run:
+          name: Build development image
+          command: make build
+      - run:
+          name: Check built image availability
+          command: docker images "funmooc"
+      - run:
+          name: Check version.json file
+          command: docker run --rm funmooc:${CIRCLE_SHA1}production cat version.json
+      # Since we cannot rely on CircleCI's Docker layers cache (for obscure
+      # reasons some subsequent jobs will benefit from a previous job cache and
+      # some others won't), we choose to save built docker images in cached
+      # directories. This ensures that we will be able to load built docker
+      # images in subsequent jobs.
+      - run:
+          name: Store docker images in cache
+          command: |
+            mkdir -p docker/images && \
+            docker save \
+              -o docker/images/funmooc.tar \
+              funmooc:${CIRCLE_SHA1}production \
+              funmooc:${CIRCLE_SHA1}development
+      - save_cache:
+          paths:
+            - ~/fun/docker/images/
+          key: docker-images-{{ .Revision }}
+
+  check-back:
+    machine: true
+    working_directory: ~/fun
+    environment:
+      BUILD_TARGET: production
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - docker-images-{{ .Revision }}
+      - <<: *upgrade-docker-compose
+      - run:
+          name: Load images to docker engine
+          command: |
+            docker load < docker/images/funmooc.tar
+      # ElasticSearch configuration
+      #
+      # We need to increase the VM max memory size, or else, ElasticSearch (ES)
+      # service won't bootstrap.
+      #
+      # Source:
+      # https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-cli-run-prod-mode
+      - run:
+          name: Increase VM max memory size for ES
+          command: |
+            sudo sysctl -w vm.max_map_count=262144
+            sudo sysctl vm.max_map_count
+      # Run back-end (Django) check
+      #
+      # Nota bene: to run the django app, we need to ensure that both PostgreSQL
+      # and ElasticSearch services are up and ready. To achieve this, we wrap
+      # commands execution with dockerize, a tiny tool installed in the CI
+      # image. In our case, dockerize will wait up to one minute that both the
+      # database and elastisearch containers opened their expected tcp port
+      # (5432 and 9200 resp.).
```

Comment:
All this comment is I think wrong. You are not using `dockerise` with `BUILD_TARGET` set to `production`. And we don;t have a `CI` image.

Reply:
I nearly commented the same but in fact I now think he is doing what this comment says, only via the make file.

Reply:
In fact, @lunika is right: this is why I have added a `sleep 10` command :sweat_smile: